library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.kt

```diff
@@ -4,38 +4,18 @@ import android.content.Context
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView
-import com.getstream.sdk.chat.view.MessageListView
-import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
 import com.getstream.sdk.chat.view.MessageListView.BubbleHelper
-import com.getstream.sdk.chat.view.MessageListView.GiphySendListener
-import com.getstream.sdk.chat.view.MessageListView.MessageClickListener
-import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
-import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
-import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
 import com.getstream.sdk.chat.view.MessageListViewStyle
 import io.getstream.chat.android.client.models.Channel
 
 class MessageListItemAdapter @JvmOverloads constructor(
     private var context: Context,
+    var viewHolderFactory: MessageViewHolderFactory,
```

Comment:
Do we want it mutable?

Reply:
One of our customization examples in the tutorial is subclassing the factory and replacing it, so I think we want to support swapping it out.

Reply:
Made this and other ctor properties of the adapter read-only, so now customization can only happen before the adapter instance is created.